pkg/controller/inference/playground_controller.go

```diff
@@ -119,6 +119,34 @@ func (r *PlaygroundReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *PlaygroundReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	mapFn := func(ctx context.Context, obj client.Object) []ctrl.Request {
+		logger := log.FromContext(ctx)
+
+		namespace := obj.GetNamespace()
+		name := obj.GetName()
+
+		playgrounds := &inferenceapi.PlaygroundList{}
+		err := r.List(ctx, playgrounds, client.InNamespace(namespace))
+		if err != nil {
+			logger.Error(err, "failed to list playgrounds")
+			return nil
+		}
+
+		var reqs []ctrl.Request
+		for _, playground := range playgrounds.Items {
+			if playground.Spec.ModelClaim != nil && playground.Spec.ModelClaim.ModelName == coreapi.ModelName(name) {
+				reqs = append(reqs, ctrl.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: namespace,
+						Name:      playground.Name,
+					},
+				})
+			}
+			// TODO: handle MultiModelsClaims in the future.
+		}
+		return reqs
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&inferenceapi.Playground{}).
 		Watches(&inferenceapi.Service{}, &handler.EnqueueRequestForObject{},
```

Comment:
I guess this can be simplified by 
```
			builder.WithPredicates(predicate.Funcs{
				CreateFunc: func(e event.CreateEvent) bool {
					return true
				},
                        }
```

This is enough I think. By the way, we need an integration test.

Reply:
If a function is nil,  the object will always be added into the queue

```go
// Update implements Predicate.
func (p TypedFuncs[T]) Update(e event.TypedUpdateEvent[T]) bool {
	if p.UpdateFunc != nil {
		return p.UpdateFunc(e)
	}
	return true
}

```

>  By the way, we need an integration test.

will do